### ISSUE TITLE
Setup Wizard

### ISSUE THREAD

**Issue by mrusme** [twostairs/paperwork#234]
It would be very neat if we could provide users some sort of setup wizards (similar to the ones WordPress and Drupal have), which would assist them with installing their Paperwork instance. It probably should be some script that's totally out-of-context and not using any of the stuff one would need composer for. By that, we could allow users running it without users having executed composer or gulp at all. This setup.php they could call via browser could even do all these steps automatically for the user, as soon as the user tells some details about his environment (where the database lies, whether he's using an apache or a nginx, etc.).

The setup.php would basically check everything - from path checks up to package suggestions. I'm not sure though whether the httpd-process this script would run inside would actually have the rights to do so.

This idea came up in #217, when @jamesrascal suggested an enhancement for the admin page.

**Comment by Baldrs** [comment]
When Drupal installer fails to find some component or does not agree with some configuration it is just refusing to continue, so it does need any rights, but provides user with information how to fix the issues.  It can even pull in composer and install, but what about gulp? You can't just give php script root rights and say do whatever you want. 

However, we can provide kind of prebuilt release, which will:

1. Have scripts and styles pre-built. No need for any rights.
2. Have composer dependencies installed. Neither need for composer.
3. Have current-version database dump made by migrations. 
4. Will provide user-friendly configuration on top of this.
5. Will be able to pull in the package with next version if available. 
6. Will halt work of the main app until removed or inaccessible.

**Comment by DeviantEng** [comment]
I feel that this is definitely needed.

My thoughts would be (I'm more of a devops guy, fyi) that when loading the "setup.php", the first step would be a prereq checker to verify all dependencies where installed, permissions were set (paperwork files are owned by the user running httpd, nginx, etc), etc, and only after all those prereqs are satisfied will you be able to click the install button.

Also, I think it is hugely important to be able to set db settings this way.  Manually having to go into the project to edit a file to set db settings doesn't seem like the ideal way.

**Comment by Liongold** [comment]
I'd like to have a look at this, if no one is planning to do it.

**Comment by Baldrs** [comment]
I've been considering to write it this weekend, but I'm not sure if I'll have time to start. I actually wanted to implement the list I wrote above.

**Comment by Liongold** [comment]
Do you want to do this yourself then? If so, did you do something or not
yet?

On Friday, February 13, 2015, Arseniy Lozicki <notifications@github.com>
wrote:

> I've been considering to write it this weekend, but I'm not sure if I'll
> have time to start. I actually wanted to implement the list I wrote above.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/twostairs/paperwork/issues/234#issuecomment-74243673>.
>


-- 
Sincerely,
Jean Spiteri.

**Comment by Baldrs** [comment]
@Liongold no, I will unlikely have enough time in nearest two weeks

**Comment by Liongold** [comment]
I thought a little about this. Before it can be done we need to have a definite format for build distribution? What are we going to include (or not in the normal user build)?

**Comment by paxperscientiam** [comment]
I think this is an excellent idea. After much digging, I came to realize that I had to edit app/config/database.php to include a reference to unix socket.

I edited the Apache wiki to reflect this:
https://github.com/twostairs/paperwork/wiki/Using-Apache-in-place-of-Nginx

**Comment by jamesrascal** [comment]
I think this issue can now be closed.

**Comment by Liongold** [comment]
I'll leave it open since some work on the wizard is still needed.

**Comment by Liongold** [comment]
The setup wizard has been rewritten. I'll close this issue now. If anyone encounters any issues with this or any other part of Paperwork please open a new issue.
